cli/generator-op-microservice/generators/microservice/templates/rest.service.ts

```diff
@@ -0,0 +1,166 @@
+import methodOverride from 'method-override';
+<% if (dbSupport) { -%>
+import mongoose from 'mongoose';
+<% } -%>
+
+import <%= apiName %>Router from './src/api';
+
+const httpsOptions = {
+  key: fs.readFileSync('./certs/server.key'),
+  cert: fs.readFileSync('./certs/server.crt')
+};
+
+const { port = 8080 } = { port : process.env.PORT };
+
+/**
+ * The <%= serviceClassName %>
+ *
+ * @class <%= serviceClassName %>
+ */
+export class <%= serviceClassName %> {
+
+  /**
+   * The express application.
+   * @type {Application}
+   */
+  public app: any;
+  public server: any;
+  /**
+   * Bootstrap the application.
+   *
+   * @class <%= serviceClassName %>
+   * @method bootstrap
+   * @static
+   */
+  public static bootstrap(): <%= serviceClassName %> {
+    return new <%= serviceClassName %>();
+  }
+
+  /**
+   * Constructor.
+   *
+   * @class <%= serviceClassName %>
+   * @constructor
+   */
+  constructor() {
+    // configure expressjs application
+    this.middleware();
+
+    // add routes
+    this.routes();
+  }
+
+  /**
+   * Configure application
+   *
+   * @class <%= serviceClassName %>
+   * @method middleware
+   */
+  public middleware() {
+    this.app = express();
+<% if (dbSupport) { -%>
+
+    /* Configuring Mongoose */
+    mongoose.plugin((schema: any) => { schema.options.usePushEach = true; });
+    mongoose.set('useNewUrlParser', true);
+    mongoose.set('useFindAndModify', false);
+    mongoose.set('useCreateIndex', true);
+    mongoose.set('useUnifiedTopology', true);
+
+    /* Establishing mongodb connection */
+    const dbCredentials = (process.env.DB_USER && process.env.DB_PASSWORD)
+      ? `${process.env.DB_USER}:${process.env.DB_PASSWORD}@`
+      : '';
+    const dbConnection = `mongodb://${dbCredentials}${process.env.DB_PATH}/${process.env.DB_NAME}`;
+
+    mongoose.connect(dbConnection, { useNewUrlParser: true, useCreateIndex: true }).catch(console.error);
+
+    mongoose.connection.on('error', error => {
+      console.error(error);
+    });
+<% } -%>
+
+    // morgan middleware to log HTTP requests
+    this.app.use(logger('":method :url :status :res[content-length] - :response-time ms :referrer :user-agent"'));
+
+    // use json form parser middleware
+    this.app.use(bodyParser.json({ limit: 1024 * 1024 * 20, type: 'application/json' }));
+
+    // use query string parser middleware
+    this.app.use(bodyParser.urlencoded({
+      extended: true
+    }));
+
+    //  catch 404 and forward to error handler
+    this.app.use(function (err: any, req: express.Request, res: express.Response, next: express.NextFunction) {
+      err.status = 404;
+      next(err);
+    });
+
+    // Mount cookie parker
+    this.app.use(cookieParser(process.env.COOKIE_PARSER));
+
+    // Mount override
+    this.app.use(methodOverride());
+
+    // Error handling
+    this.app.use(errorHandler());
+
+    this.app.set('port', port);
+
+    this.server = https.createServer(httpsOptions, this.app);
```

Comment:
@riginoommen Are the tests passing for https server here?
Or should there be a test environment exception for http server here?

Reply:
Yes

Reply:
Then this needs to be http server for test environment.

Reply:
sorry i got confused https is passing